internal/server/auditsink/auditsink.go

```diff
@@ -0,0 +1,97 @@
+package auditsink
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// AuditEvent holds information that represents an audit internally.
+type AuditEvent struct {
+	ResourceName string `json:"resourceName"`
+}
+
+// AuditSink is the abstraction for various audit sink configurations
+// that Flipt will support.
+type AuditSink interface {
+	SendAudits([]AuditEvent) error
+	fmt.Stringer
+}
+
+// Publisher holds information about the configured sinks that we are going
+// to send audit events to.
+type Publisher struct {
+	mtx         sync.Mutex
+	logger      *zap.Logger
+	bufferSize  int
+	sinks       []AuditSink
+	auditEvents []AuditEvent
+	ticker      *time.Ticker
+}
+
+// NewPublisher is the constructor for a Publisher.
+func NewPublisher(logger *zap.Logger, bufferSize int, sinks []AuditSink, tickerDuration time.Duration) *Publisher {
+	p := &Publisher{
+		logger:      logger,
+		bufferSize:  bufferSize,
+		sinks:       sinks,
+		auditEvents: make([]AuditEvent, 0),
+		ticker:      time.NewTicker(tickerDuration),
+	}
+
+	go p.flushWhenNecessary()
+
+	return p
+}
+
+// flush flushes the buffer to the configured sinks.
+func (p *Publisher) flush(auditEvent *AuditEvent) {
+	copiedEvents := make([]AuditEvent, len(p.auditEvents))
+	copy(copiedEvents, p.auditEvents)
+	p.auditEvents = make([]AuditEvent, 0)
```

Comment:
Once you've safely copied the events into a new slice you can forgo allocating and creating another here by just re-slicing your original audit events:
```suggestion
	p.auditEvents = p.auditEvents[:0]
```

This will avoid constantly allocating and re-growing your buffer.